### PR TITLE
get latest nightly build files from given base url

### PR DESCRIPTION
**Pull request by OscarQQ**
related: backlog issue 542
Signed-off-by: Yixin Qian <Yixin.Qian@ibm.com>

### CODE REVIEW

**Comment by OscarQQ** [comment]
run aqa --sdk_resource nightly --build_list functional --target floatSanityTests --platform x86-64_linux --customized_sdk_url https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Nightly/

**Comment by OscarQQ** [comment]
run aqa --sdk_resource nightly --build_list functional --target testSCCMLTests2 --platform x86-64_windows --customized_sdk_url https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Nightly/

**Comment by OscarQQ** [comment]
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/1000/consoleFull
getting tar.gz given the base url https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Nightly/

**Comment by llxia** [comment]
Regarding run aqa failure (https://github.com/adoptium/aqa-tests/pull/2695#issuecomment-871089710), issue opened at https://github.com/adoptium/aqa-tests/issues/2697

**Comment by llxia** [comment]
Thanks @OscarQQ . With this PR, we will allow users to provide a **base artifactory URL** (URL before the build number). Our test script will try to figure out the latest build number and it will try to download all tar (including SDK, test image, debug image, etc).

Use case in Grinder:
SDK_RESOURCE=nightly
CUSTOMIZED_SDK_URL=https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Nightly

**Comment by llxia** [comment]
@OscarQQ users may or may not have `/` at the end of the URL, please make sure we cover both cases. That is, https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/1001/console should work. Thanks

File: get.sh [hunk 1]
```diff
@@ -177,7 +177,29 @@ getBinaryOpenjdk()
 		fi
 	fi
 
-	if [ "$CUSTOMIZED_SDK_URL" != "" ]; then
+	if [ "$SDK_RESOURCE" == "nightly" ] && [ "$CUSTOMIZED_SDK_URL" != "" ]; then
+		if [ "$USERNAME" != "" ] && [ "$PASSWORD" != "" ]; then
+			curl_options="--user $USERNAME:$PASSWORD"
+		fi
```
Comment (llxia) [inline get.sh:183]: Please move the username/password outside of the if, so both line 180 and line 202 can use it.

**Comment by OscarQQ** [comment]
Grinder/16767/  x86-64_windows  internal link
Grinder/16770/  ppc64_aix  internal link
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/1030/  x86-64_linux. external link

File: get.sh [hunk 2]
```diff
@@ -177,13 +177,33 @@ getBinaryOpenjdk()
...
+	# if these are passed through via withCredentials(CUSTOMIZED_SDK_URL_CREDENTIAL_ID) these will not be visible within job output,
+	# if supplied when run manually with --username and --password these will be seen in plaintext within job output
+	if [ "$USERNAME" != "" ] && [ "$PASSWORD" != "" ]; then
+		curl_options="--user $USERNAME:$PASSWORD"
+	fi
+
+	if [ "$SDK_RESOURCE" == "nightly" ] && [ "$CUSTOMIZED_SDK_URL" != "" ]; then
 		if [ "$USERNAME" != "" ] && [ "$PASSWORD" != "" ]; then
 			curl_options="--user $USERNAME:$PASSWORD"
 		fi
```
Comment (llxia) [inline get.sh:189]: Line 187- 189 should be removed

**Review by llxia — APPROVED** [review]
LGTM. Thanks a lot @OscarQQ 
During the testing, we noticed that the latest build may not contain SDK (due to SDK build failure). We will add further enhancement in future PR - try old builds if the latest build does not contain SDK.

**Review by smlambert — APPROVED** [review]
thanks for working on this enhancement @OscarQQ !  LGTM

**Pull request merged**